src/read.c

```diff
@@ -1830,10 +1838,8 @@ static avifResult avifParseItemLocationBox(avifMeta * meta, const uint8_t * raw,
         uint16_t extentCount;                                                                // unsigned int(16) extent_count;
         AVIF_CHECKERR(avifROStreamReadU16(&s, &extentCount), AVIF_RESULT_BMFF_PARSE_FAILED); //
         for (int extentIter = 0; extentIter < extentCount; ++extentIter) {
-            // If extent_index is ever supported, this spec must be implemented here:
-            // ::  if (((version == 1) || (version == 2)) && (index_size > 0)) {
-            // ::      unsigned int(index_size*8) extent_index;
-            // ::  }
+            uint64_t itemReferenceIndex; // unsigned int(index_size*8) item_reference_index; (ignored unless construction_method=2)
+            AVIF_CHECKERR(avifROStreamReadUX8(&s, &itemReferenceIndex, indexSize), AVIF_RESULT_BMFF_PARSE_FAILED);
```

Comment:
Nit/optional: It would be clearer to put this call inside `if (indexSize > 0)`. The corresponding spec is:

```
    if (((version == 1) || (version == 2)) && (index_size > 0)) {
         unsigned int(index_size*8) item_reference_index;
    }
```

But perhaps it is that spec that should omit the `&& (index_size > 0)` check.

Reply:
Thanks. While reviewing https://github.com/AOMediaCodec/libavif/pull/2091, I discovered a subtle point about `item_reference_index` -- its value is 1 rather than 0 when `item_size` is 0. Here is the excerpt from ISO/IEC 14496-12:2022 Section 8.11.3.1, page 82:

> The `item_reference_index` is only used for the method `item_offset`; it indicates the 1-based index of the item reference with `referenceType 'iloc'` linked from this item. If `index_size` is 0, then the value 1 is implied; the value 0 is reserved.

Calling `avifROStreamReadUX8(&s, &itemReferenceIndex, indexSize)` with `indexSize=0` will set `itemReferenceIndex` to 0. So `item_reference_index` needs different handling anyway.